backend/routes/middlewares/index.js

```diff
@@ -88,22 +88,33 @@ function checkProject(req, res, next) {
 
 function checkBadState(req, res, next) {
   const projectId = JSON.parse(req.body.message).projectId;
-
+  console.log('herrreeee')
   // get the latest record
+  try{
   Proposal.find({ projectId })
     .sort({ round: -1 })
     .limit(1)
     .exec(async (err, data) => {
+      console.log('herrreeee')
       if (err) {
         return res.status(400).send(err);
       }
-      if (!data[0]) {
+      console.log('herrreeee')
+      if (!data || data[0]) {
         return next();
```

Comment:
I don't know if this check is correct...

I'm pretty sure that the line below `if (data[0].delivered.status !== 2) {` is ever going to be executed because the check `if (!data || data[0]) {` will always be valid.

Try the code below to see what I mean.
`console.log("=======================\nBoolean check")
let data = false;
if (!data) console.log("Data is false");
if (data) console.log("Data is valid");

console.log("=======================\nArray check")
data = [1];
if (!data) console.log("Data is false");
if (data) console.log("Data is valid");
if (data[0]) console.log("Data at index 0 is:", data[0]);

console.log("=======================\nNull check")
data = null;
if (!data) console.log("Data is false");
if (data) console.log("Data is valid");
if (data && data[0]) console.log("Data at index 0 is:", data[0]);
if (data === null) console.log("Data is null");

console.log("=======================\nObject check")
data = [{
    delivered: {
        index: 1,
        status: 1
    }
},{
    delivered: {
        index: 2,
        status: 2
    }
},]

if (!data) console.log("Data is false");
if (data) console.log("Data is valid");
if (data[0]) console.log("Data at index 0 is:", data[0]);
if (data[0].delivered.status !== 2) console.log("Data at index 0 is not equal to 2 => Data at index 0 is...:", data[0].delivered.status);
if (data[1].delivered.status !== 2) console.log("Data at index 1 is not equal to 2 => Data at index 1 is...:", data[1].delivered.status);
`

Reply:
@trizin is going to also review this, and then we can push the update.

Reply:
True, `data` will never be `undefined` and we should only check if the project has proposals.
`if (!data[0]) {` is sufficient here